### PR TITLE
Adds getKey option to passphrase encryption

### LINKED ISSUE #8928: Add getKey as an argument to encryptMessageWithPassword
### Description

Update `encryptMessageWithPassword` to take `getKey` function as an argument, so we can use options.getKey if provided for the key.

### Acceptance Criteria

- All the unit tests should pass
- Add new test case if needed

### Additional Information

- LIP PR https://github.com/LiskHQ/lips/pull/465/

### PR DESCRIPTION
**Pull request by has5aan**
### What was the problem?

This PR resolves #8928 

### How was it solved?

Added a new optional `getKey` parameters to encrypt and decrypt methods to take precedence for key retrieval.

### How was it tested?

Implemented unit tests.

### CODE REVIEW

File: elements/lisk-cryptography/src/encrypt.ts [hunk 1]
```diff
@@ -85,6 +85,13 @@ export const encryptAES128GCMWithPassword = async (
 			iterations?: number;
 			memorySize?: number;
 		};
+		getKey?: (options: {
+			password: string;
```
[the remark this answers is not in the document]
Comment (has5aan) [reply elements/lisk-cryptography/src/encrypt.ts:89]: Updated under the [commit](https://github.com/LiskHQ/lisk-sdk/pull/8930/commits/53b5652b23d293cbd250874bbbc5a96ee976a52c).

**Review by ishantiw — CHANGES_REQUESTED** [review]

File: elements/lisk-cryptography/test/encrypt.spec.ts [hunk 2]
```diff
@@ -216,6 +235,29 @@ describe('encrypt', () => {
 					decryptMessageWithPassword(encryptedMessage, defaultPassword, 'utf-8'),
 				).rejects.toThrow('Unsupported state or unable to authenticate data');
 			});
+
+			it('should call options.getKey if provided', async () => {
```
[the remark this answers is not in the document]
Comment (has5aan) [reply elements/lisk-cryptography/test/encrypt.spec.ts:239]: One test case for encrypt and this one is for the decrypt function, to test that the function takes precedence.

[the remark this answers is not in the document]
Comment (ishantiw) [reply elements/lisk-cryptography/test/encrypt.spec.ts:239]: ah i missed out, i thought its in the same test suite

**Review by ishantiw — APPROVED** [review]
LGTM

**Pull request merged**